### PR TITLE
[saas-file-owners] add compare diffs as comment to merge request

### PR DESCRIPTION
**Pull request by maorfr**
this PR adds a comment to MRs updating saas files with the compare url for each of the saas file diffs in the MR.

this will allow us and our users to quickly review what is currently being promoted.

### CODE REVIEW

**Comment by arilivigni** [comment]
LGTM

**Review by aditya-konarde — APPROVED** [review]
Lgtm!

File: reconcile/saas_file_owners.py [hunk 1]
```diff
@@ -99,6 +101,28 @@ def collect_baseline():
...
+            if d['environment'] != c['environment']:
+                continue
+            if d['cluster'] != c['cluster']:
+                continue
+            if d['namespace'] != c['namespace']:
+                continue
+            if d['ref'] == c['ref']:
+                continue
+            compare_diffs.add(
+                f"{d['url']}/compare/{c['ref']}...{d['ref']}")
```
Comment (rporres) [inline reconcile/saas_file_owners.py:121]: Cool that this works in gitlab and github :)

File: reconcile/saas_file_owners.py [hunk 2]
```diff
@@ -226,6 +250,14 @@ def run(dry_run, gitlab_project_id=None, gitlab_merge_request_id=None,
     current_state = baseline['state']
     desired_state = collect_state()
     diffs = [s for s in desired_state if s not in current_state]
+
+    compare_diffs = collect_compare_diffs(current_state, desired_state)
+    compare_diffs_comment_body = '\n'.join([f'- {d}' for d in compare_diffs])
```
Comment (rporres) [inline reconcile/saas_file_owners.py:255]: nit: I guess you can test on `compare_diffs` and join inside once you know you have something to join 😬

**Review by rporres — COMMENTED** [review]
Great addition! A bit of nit picking and the usual call for documentation :-D

File: reconcile/saas_file_owners.py (hunk 1 above)
Comment (maorfr) [reply reconcile/saas_file_owners.py:121]: i was surprised!

File: reconcile/saas_file_owners.py [hunk 3]
```diff
@@ -99,6 +101,28 @@ def collect_baseline():
     return {'owners': owners, 'state': state}
 
 
+def collect_compare_diffs(current_state, desired_state):
```
[the remark this answers is not in the document]
Comment (maorfr) [reply reconcile/saas_file_owners.py:104]: done!

File: reconcile/saas_file_owners.py (hunk 2 above)
Comment (maorfr) [reply reconcile/saas_file_owners.py]: done!

**Pull request merged**